### PR TITLE
chore: separate launcher(ARTP1353)

### PR DESCRIPTION
**Pull request by slcfv3**
chore: separate launcher(ARTP1353)
-separate the standalone launcher from the Reactive Trader
-migrate all the code and dependencies from the RT main directory to the launcher directory
-launcher app can run independently in the launcher directory now

### CODE REVIEW

File: src/launcher/standalone_launcher/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+node_modules
+npm-debug.log
+Dockerfile
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/.dockerignore:4]: Add newline at end of file. (Please set up your IDE to insert these automatically).

File: src/launcher/standalone_launcher/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+indent_style = space
+indent_size = 2
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/.editorconfig:12]: Add newline at end of file. (Please set up your IDE to insert these automatically).

File: src/launcher/standalone_launcher/.env.development [hunk 3]
```diff
@@ -0,0 +1,9 @@
+REACT_APP_BROKER_HOST=web-dev.adaptivecluster.com
+REACT_APP_BROKER_PORT=80
+REACT_APP_SYMPHONY_APP_ID=reactiveTrader-dev
+ENVIRONMENT_NAME=local
+
+# The REACT_APP_ANALYTICS_URL var is only used from the OF launcher, so most of the times we'll want to 
+# launch the live app even when doing dev work on the launcher itself. Otherwise switch to the localhost one
+REACT_APP_ANALYTICS_URL=https://demo-reactive-analytics.adaptivecluster.com/
+# REACT_APP_ANALYTICS_URL=http://localhost:3005
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/.env.development:9]: Add newline at end of file. (Please set up your IDE to insert these automatically).

File: src/launcher/standalone_launcher/.env.production [hunk 4]
```diff
@@ -0,0 +1 @@
+REACT_APP_GREENKEY_URL=wss://web-beta.adaptivecluster.com/client/ws/speech
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/.env.production:1]: Add newline at end of file. (Please set up your IDE to insert these automatically).

File: src/launcher/standalone_launcher/.storybook/preview.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+export const parameters = {
+  actions: { argTypesRegex: "^on[A-Z].*" },
+}
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/.storybook/preview.js:4]: Add newline at end of file. (Please set up your IDE to insert these automatically).

File: src/launcher/standalone_launcher/README.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
+# Reactive Trader Client
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/README.md:1]: This README is not relevant to the launcher. We need to write a specific README for the launcher. Please delete anything in here that is only related to the client app.

File: src/launcher/standalone_launcher/install/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
+# Installers
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/install/README.md:1]: Delete the `install` directory for now. We have all the installers in src/client at the moment. We will revisit this later.

File: src/launcher/standalone_launcher/lib/README.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# lib
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/lib/README.md:1]: Delete this file and `downloadInstaller.js`. We need to revisit where the installers live after this PR.

File: src/launcher/standalone_launcher/nginx/nginx.conf [hunk 9]
```diff
@@ -0,0 +1,63 @@
+# nginx.conf  --  docker-openresty
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/nginx/nginx.conf:1]: Delete this file. We have this in client. We'll lift nginx and all the Docker configuration out later.

File: src/launcher/standalone_launcher/package.json [hunk 10]
```diff
@@ -0,0 +1,166 @@
+{
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/package.json:1]: Please go through this and check which dependencies are actually used by launcher. We won't need everything here.

File: src/launcher/standalone_launcher/package.json [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+  "name": "standalone-launcher",
+  "description": "Standalone Launcher, laucher of reactive apps.",
```
Comment (garyadaptive) [inline src/launcher/standalone_launcher/package.json:3]: Typo: s/laucher/launcher

**Review by garyadaptive — CHANGES_REQUESTED** [review]
Thanks for this. We'll need to tighten this up before we can merge in.

High level action points:
* Lift out all non-launcher code. Many components and tests have been copied across that are not relevant - e.g. spot tile and blotter components under MainRoute.
* We're pulling in a lot of dependencies that are also not required. Once the redundant code has been removed, we'll need to clean up package.json to only pull in launcher dependencies.
* Let's flatten the directory: src/launcher/standalone_launcher can just be src/launcher.
* Anything relating to Docker, nginx or installers can be deleted for now. We already have these under src/client and I would like to refactor this as a separate piece of work.

There will be some other work around OpenFin manifests and port collisions between the client app and launcher app, but it's currently difficult to review with all the extraneous code.

File: src/launcher/standalone_launcher/package.json (hunk 11 above)
Comment (slcfv3) [reply src/launcher/standalone_launcher/package.json]: chore: PR Clean UP(ARTP1353)

Remove the redundant codes
Remove the unused dependencies
Flatten the directory
Delete the docker and ngnix related directories.

**Comment by garyadaptive** [comment]
Let's restore the launcher code inside src/client before we merge. I'm happy for the standalone launcher code to be on master, but we won't be ready to switch over when this is merged in. It will be safer to deprecate than delete.

Also, "refactor:" is probably the most appropriate commit type for your changes. Please review https://www.conventionalcommits.org/en/v1.0.0/ for recommendations on commit type usage.

**Comment by garyadaptive** [comment]
Let's restore the launcher code inside src/client before we merge. I'm happy for the standalone launcher code to be on master, but we won't be ready to switch over when this is merged in. It will be safer to deprecate than delete.

Also, "refactor:" is probably the most appropriate commit type for your changes. Please review https://www.conventionalcommits.org/en/v1.0.0/ for recommendations on commit type usage.

**Pull request closed without merging**